docs/reference/triggers/kafka.md

```diff
@@ -1,110 +1,315 @@
+## Configuration parameters
+<!-- See https://kafka.apache.org/documentation/#consumerconfigs + types and
+  default values in pkg/processor/trigger/kafka/types.go. -->
+
+Use the following trigger attributes for basic configurations of your Kafka trigger.
+You can configure each attribute either in the `triggers.<trigger>.attributes.<attribute>` function `spec` element (for example, `triggers.myKafkaTrigger.attributes.sessionTimeout`) or by setting the matching `nuclio.io` annotation key (for example, `nuclio.io/kafka-session-timeout`); (note that not all attributes have matching annotation keys).
+For more information on Nuclio function configuration, see the [function-configuration reference](/docs/reference/function-configuration/function-configuration-reference.md).
+
+> **Note:** For more advanced configuration parameters, see the configuration sections under [How a message travels through Nuclio to the handler](#message-course-config-params) and [Rebalancing](#rebalancing-config-params). For an example, see [Configuration example](#config-example).
+
+- <a id="topic"></a>**`topic`** - The name of the topic on which to listen.
+  <br/>
+  **Type:** `string`
+
+- <a id="brokers"></a>**`brokers`** - A list of broker IP addresses.
+  <br/>
+  **Type:** `[]string`
+
+- <a id="paritions"></a>**`paritions`** - A list of partitions for which the function receives events.
+  <br/>
+  **Type:** `[]int`
+
+- <a id="consumerGroup"></a>**`consumerGroup`** - The name of the Kafka consumer group to use.
+  <br/>
+  **Type:** `string`
+
+- <a id="initialOffset"></a>**`initialOffset`** - The location (offset) within the partition at which to begin the message processing.
+  Currently, you can begin the processing either with the earliest or latest ingested messages.
+  <br/>
+  **Type:** `string`
+  <br/>
+  **Valid Values:** `"earliers" | "latest"`
+  <br/>
+  **Default Value:** `"earliest"`
+
+- <a id="sasl"></a>**`sasl`** - A simple authentication and security layer (SASL) object.
+  <br/>
+  **Type:** `object` with the following attributes -
+
+  - **`enable`** (`bool`) - Enable authentication.
+  - **`user`** (`string`) - Username to be used for authenticatio.
+  - **`password`** (`string`) - Password to be used for authentication.
+
+- <a id="sessionTimeout"></a>**`sessionTimeout`** (`kafka-session-timeout`) - The timeout used to detect consumer failures when using Kafka's group management facility. The consumer sends periodic heartbeats to indicate its liveness to the broker. If no heartbeats are received by the broker before the expiration of this session timeout, the broker removes this consumer from the group and initiates rebalancing. Note that the value must be in the allowable range, as configured in the `group.min.session.timeout.ms` and `group.max.session.timeout.ms` broker configuration parameters.
+  <br/>
+  **Type:** `time.Duration` (`string` for annotation)
+  <br/>
+  **Default Value:** `10 * time.Second` (10 seconds)
```

Comment:
Default duration values are in the Go duration format - `3s`, `100ms` etc. We should keep this so that users know the format they need to populate, not `10 * time.Second` which is the "parsed" version of the duration

Reply:
I saw in the code that for duration fields we have both `xxxYyy` variables of type `time.Duration` (e.g., `sessionTimeout`) — which are initialized using `time` (e.g., `10 * time.Second`) — and `XxxYyy` variables of type `string` (e.g., `SessionTimeout`) — which are used for annotations. Are you saying that the duration values in the function `spec` configurations should also be of type `string`, like in the annotations, and `timeDuration` is only used internally?

Reply:
`SessionTimeout` is a string, populated by the user (e.g. `3s`)
`sessionTimeout` is an internal field of type time.Duration that receives the parsed output of `SessionTimeout`. Therefore the user only sees the string fields and must populate according to the Go format.

This pattern repeats itself across many configuration parameters.

Reply:
As we discussed, I changed the type description for all duration string configurations as follows, and I changed the default values to `"<n><unit>"` (e.g., `"5s"`) — commit f24e603:
>**Type:** `string` &mdash; a string containing one or more duration strings of the format `"[0-9]+[ns|us|ms|s|m|h]"`; for example, `"300ms"` (300 milliseconds) or `"2h45m"` (2 hours and 45 minutes). See the [`ParseDuration`](https://golang.org/pkg/time/#ParseDuration) Go function.